org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/ui/desktop/outline/AbstractOutline.java

```diff
@@ -757,7 +761,21 @@ private void handleActivePageChanged(IPage<?> deselectedPage, IPage<?> selectedP
 
   @Override
   public void setDefaultDetailForm(IForm form) {
+    IForm previousForm = getDefaultDetailForm();
+    if (form == previousForm) {
+      return;
+    }
+    if (previousForm != null) {
+      previousForm.removeFormListener(m_defaultDetailFormDisposeListener);
+      m_defaultDetailFormDisposeListener = null;
+    }
     if (form != null) {
+      m_defaultDetailFormDisposeListener = e -> {
+        if (e.getType() == FormEvent.TYPE_CLOSED) {
+          setDefaultDetailForm(null);
+        }
+      };
+      form.addFormListener(m_defaultDetailFormDisposeListener);
```

Comment:
Should we add the other options (like in Outline.js#_adaptDefaultDetailForm) here as well (e.g. modality)?

Reply:
This should not be necessary, since the ui already does it. Also, in Java, modal is automatically set to false for views.